jykuo-love-shiritori/twp
# [frontend] Coupon api

Reeeeeeeeesolves #94 
> [!IMPORTANT] 
> see the edited test approach and the additional context

### Changes

#### Page
- [x] [/admin/manageCoupon](http://localhost:8080/admin/manageCoupons)
- [x] /admin/manageCoupon/couponID
- [x] [/admin/manageCoupon/new](http://localhost:8080/admin/manageCoupons/new)
- [x] [/seller/manageCoupon](http://localhost:8080/user/seller/manageCoupons)
- [x] [/seller/manageCoupon/new](http://localhost:8080/user/seller/manageCoupons/new)
- [x] /seller/manageCoupon/couponID
- [x] /sellerCoupon

#### APIs
- Admin Get Coupon
- Admin Add Coupon
- Admin Get Coupon Detail
- Admin Delete Coupon
- Admin Edit Coupon
- Seller get shop coupon
- Seller add coupon
- Seller get coupon detail
- Seller delete coupon
- Seller edit coupon
- Seller add coupon tag
- Seller delete coupon tag
- Get Shop Coupons

#### Function
- Admin/Seller can view/edit/add/delete coupons
- Seller can add/delete tags of coupons
- View shop coupon


### Testing Approach (if unconventional) **EDITED**
```sh
docker compose up --build
```
- go to the changed page
- seller
  - login with user, and make sure its shop is enabled
  - go to manage shop coupon page, try to add/edit/delete coupon
- admin
  - login with admin
  - go to manage global coupon page, try to add/edit/delete coupon
- user view shop
  - i changed the routing, so u have to know the users' name to go to their shop
  - go to ``/{user_name}/coupons`` to view the shop coupon and global coupon

### Additional context
some pages will break after the auth check. i.e. product page
and itll get fixed just not in this branch.
